src/AppInstallerCommonCore/Filesystem.cpp

```diff
@@ -62,6 +62,14 @@ namespace AppInstaller::Filesystem
         return (GetVolumeInformationFlags(path) & FILE_SUPPORTS_REPARSE_POINTS) != 0;
     }
 
+    bool PathEscapesBaseDirectory(const std::filesystem::path& path, const std::filesystem::path& base)
+    {
+        // Input for path must exist or this will throw a filesystem error.
+        const auto& pathString = std::filesystem::canonical(path).u8string();
+        const auto& baseString = base.lexically_normal().u8string();
+        return pathString.find(baseString) == std::string::npos;

```

Comment:
From the description on how `lexically_relative` is implemented, we should use `mismatch` here instead:
```
auto [a, b] = mismatch(path.begin(), path.end(), base.begin(), base.end());
return (b != base.end());
```
Which I believe ensures that the paths match at least until the end of `base`.

Reply:
Didn't know about mismatch... learned something new :) Changed to suggested.